openedx/core/djangoapps/credit/models.py

```diff
@@ -114,29 +114,47 @@ class CreditProvider(TimeStampedModel):
     CREDIT_PROVIDERS_CACHE_KEY = "credit.providers.list"
 
     @classmethod
-    def get_credit_providers(cls):
+    def get_credit_providers(cls, providers_list=None):
         """
-        Retrieve a list of all credit providers, represented
+        Retrieve a list of all credit providers or filter on providers_list, represented
         as dictionaries.
+
+        Arguments:
+            provider_list (list of strings or None): contains list of ids if required results
+            to be filtered, None for all providers.
+
+        Returns:
+            list of providers represented as dictionaries.
+
         """
-        # Attempt to retrieve the credit provider list from the cache
+        # Attempt to retrieve the credit provider list from the cache if provider_list is None
         # The cache key is invalidated when the provider list is updated
         # (a post-save signal handler on the CreditProvider model)
         # This doesn't happen very often, so we would expect a *very* high
         # cache hit rate.
-        providers = cache.get(cls.CREDIT_PROVIDERS_CACHE_KEY)
-
-        # Cache miss: construct the provider list and save it in the cache
-        if providers is None:
-            providers = [
-                {
-                    "id": provider.provider_id,
-                    "display_name": provider.display_name,
-                    "status_url": provider.provider_status_url,
-                }
-                for provider in CreditProvider.objects.filter(active=True)
-            ]
-            cache.set(cls.CREDIT_PROVIDERS_CACHE_KEY, providers)
+
+        if providers_list is None:
+            credit_providers = cache.get(cls.CREDIT_PROVIDERS_CACHE_KEY)
+            if credit_providers is None:
+                # Cache miss: construct the provider list and save it in the cache
+
+                credit_providers = CreditProvider.objects.filter(active=True)
+                cache.set(cls.CREDIT_PROVIDERS_CACHE_KEY, credit_providers)
```

Comment:
It looks like this may be setting the cache value to a Django model, whereas before it was setting the value to a dictionary.  Django's default behavior is to pickle the object, which could be a problem if the model definition changes (for example, on a code deploy).